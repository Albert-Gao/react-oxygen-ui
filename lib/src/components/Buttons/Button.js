"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const styles_1 = require("../../styles");
const utils_1 = require("../../utils/utils");
const BaseBox_1 = require("../Box/BaseBox");
const shared_1 = require("./shared");
exports.Button = ({ noDefaultHoverCSS, children, css, disabled, takeRef, ...rest }) => {
    const cssToAdd = `
    ${shared_1.getBaseButtonCSS(noDefaultHoverCSS || disabled)};
    ${shared_1.baseNonFlatButtonCSS};
    ${styles_1.getSafeCss(css)};
  `;
    return (<BaseBox_1.BaseBox as="button" border={0} borderColor="transparent" borderRadius={0} fontSize={styles_1.DEFAULT_FONT_SIZE.bodyHeading} px={styles_1.DEFAULT_SPACE.small} minHeight="44px" maxHeight="44px" bg="primary" color="white" css={cssToAdd} {...utils_1.getRefProp(takeRef)} {...rest}>
      {children}
    </BaseBox_1.BaseBox>);
};
