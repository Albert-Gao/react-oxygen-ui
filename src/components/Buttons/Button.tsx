import React from 'react';
import {
  css as cssFunc,
  DEFAULT_FONT_SIZE,
  DEFAULT_SPACE,
  getSafeCss,
} from '../../styles';
import { getRefProp } from '../../utils/utils';
import { BaseBox } from '../Box/BaseBox';
import { IBaseBoxWithTakeRef } from '../Box/BaseBox.type';
import {
  baseNonFlatButtonCSS,
  getBaseButtonCSS,
  INoDefaultHoverCSS,
} from './shared';

export interface IButton
  extends IBaseBoxWithTakeRef,
    INoDefaultHoverCSS {
  disabled?: boolean;
}

export const Button: React.FC<IButton> = ({
  noDefaultHoverCSS,
  children,
  css,
  disabled,
  takeRef,
  ...rest
}) => {
  const cssToAdd = `
    ${getBaseButtonCSS(noDefaultHoverCSS || disabled)};
    ${baseNonFlatButtonCSS};
    ${getSafeCss(css)};
  `;

  return (
    <BaseBox
      as="button"
      border={0}
      borderColor="transparent"
      borderRadius={0}
      fontSize={DEFAULT_FONT_SIZE.bodyHeading}
      px={DEFAULT_SPACE.small}
      minHeight="44px"
      maxHeight="44px"
      bg="primary"
      color="white"
      css={cssToAdd}
      {...getRefProp(takeRef)}
      {...rest}
    >
      {children}
    </BaseBox>
  );
};
