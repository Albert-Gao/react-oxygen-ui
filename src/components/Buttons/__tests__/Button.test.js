import 'jest-styled-components';
import 'jest-dom/extend-expect';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { Button } from '../Button';
import {
  ThemeProvider,
  DEFAULT_FONT_SIZE,
  defaultTheme,
} from '../../../styles';

afterEach(cleanup);

describe('Test <Button />: ', () => {
  test('fontSize should be added', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Button fontSize={DEFAULT_FONT_SIZE.mainHeading} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();

    // the new css is there
    expect(firstChild).toHaveStyleRule(
      'font-size',
      defaultTheme.fontSizes[DEFAULT_FONT_SIZE.mainHeading],
    );
  });
});
