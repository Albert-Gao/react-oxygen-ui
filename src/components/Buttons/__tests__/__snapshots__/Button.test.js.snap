// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Button />:  fontSize should be added 1`] = `
.c0 {
  box-sizing: border-box;
  padding-left: 32px;
  padding-right: 32px;
  max-height: 44px;
  min-height: 44px;
  font-size: 48px;
  border: 0;
  border-radius: 4px;
  border-color: transparent;
  color: #FFFFFF;
  background-color: #2CAACA;
  background-color: #2CAACA;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  display: inline-block;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-align: center;
  line-height: inherit;
  white-space: nowrap;
  cursor: pointer;
  text-transform: uppercase;
  overflow: hidden;
}

.c0:hover {
  opacity: 0.9;
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c0:active {
  -webkit-transform: scale(1) translate(1px,1px);
  -ms-transform: scale(1) translate(1px,1px);
  transform: scale(1) translate(1px,1px);
}

<button
  class="c0"
  color="white"
  font-size="5"
/>
`;
